opm/simulators/flow/BlackoilModelEbos.hpp

```diff
@@ -1794,7 +1749,12 @@ namespace Opm {
             const auto& solution = ebosSimulator().model().solution(0);
 
             std::vector<int> domain_order(domains_.size());
-            if (param_.local_solve_approach_ == "gauss-seidel") {
+            switch (param_.local_solve_approach_) {
+            case DomainSolveApproach::Jacobi: {
+                std::iota(domain_order.begin(), domain_order.end(), 0);
+                break;
+            }
+            case DomainSolveApproach::GaussSeidel: {
```

Comment:
So this isn't **completely** equivalent to the original code.  That would return `[0, 1, ..., n-1]` for anything other than Gauss&ndash;Seidel.  This revised logic will return `[0, 1, ..., n-1]` only for the Jacobi method, and will return `[0, 0, ..., 0]` for anything else.  Maybe we should `throw` for unknown `local_solve_approach_` values?

Reply:
it's an extensive switch.

Reply:
And by "anything else" I mean anything else that's **not** Gauss&ndash;Seidel.  Sorry for the confusion.

Reply:
> it's an extensive switch.

It's not.  That's one of the surprises of `enum class`.  The actual enumerator values are all of the values of the underlying integer types.

Reply:
well okay, but it's still the only two values that can be assigned to the variable, we check and throw earlier in the process.

Reply:
added some default labels to retain original behavior (the other block had the opposite logic...)